embedded-bins/containerd/Dockerfile

```diff
@@ -6,13 +6,15 @@ ENV GOPATH=/go
 RUN apk upgrade -U -a && apk add \
 	build-base git \
 	btrfs-progs-dev btrfs-progs-static \
-	protoc
+	protoc curl
 
 ARG VERSION
 RUN mkdir -p $GOPATH/src/github.com/containerd/containerd
 RUN git -c advice.detachedHead=false clone -b v$VERSION --depth=1 https://github.com/containerd/containerd.git $GOPATH/src/github.com/containerd/containerd
 WORKDIR /go/src/github.com/containerd/containerd
 
+RUN curl -L https://github.com/containerd/containerd/pull/7845.patch | git apply
```

Comment:
I would use the commit instead as the pull request may change and make things unpredictable:
https://github.com/containerd/containerd/commit/cf6bb0841b6ae2cd91255a4df89aee6ea5192dc6.patch

Also I would wait for it to be merged, becuse some of the failures (including a race condition) seem legit to me, but overall I like the idea.

Reply:
Yes. We shouldn't merge this until its merged upstream. That's why I created it as a draft. But it gives us a possibility to test it early.